packages/financial-templates-lib/src/clients/OptimisticOracleClient.ts

```diff
@@ -138,39 +192,42 @@ export class OptimisticOracleClient {
     });
 
     // Store proposals that have NOT been disputed and have NOT been settled, and reformat data.
-    const undisputedProposals = proposalEvents.filter((event) => {
+    const undisputedProposals = (proposalEvents as ProposePrice[]).filter((event) => {
       const key = this._getPriceRequestKey(event);
-      const hasDispute = disputeEvents.find((disputeEvent) => this._getPriceRequestKey(disputeEvent) === key);
+      const hasDispute = (disputeEvents as DisputePrice[]).find(
+        (disputeEvent) => this._getPriceRequestKey(disputeEvent) === key
+      );
       return hasDispute === undefined;
     });
 
-    const unsettledProposals = (
-      await Promise.all(
-        undisputedProposals.map(async (event) => {
-          const state = await this.oracle.methods
-            .getState(
-              event.returnValues.requester,
-              event.returnValues.identifier,
-              event.returnValues.timestamp,
-              event.returnValues.ancillaryData ? event.returnValues.ancillaryData : "0x"
-            )
-            .call();
-
-          // For unsettled proposals, reformat the data:
-          if (state !== OptimisticOracleRequestStatesEnum.SETTLED) {
-            return {
-              ...event.returnValues,
-              identifier: this.hexToUtf8(event.returnValues.identifier),
-              ancillaryData: event.returnValues.ancillaryData ? event.returnValues.ancillaryData : "0x",
-            };
-          }
-        })
-      )
-    ).filter(isDefined);
+    const unsettledProposals = undisputedProposals
+      .map((event) => {
+        const key = this._getPriceRequestKey(event);
+        const settlement = settleEvents.find((settleEvent) => this._getPriceRequestKey(settleEvent) === key);
+        if (settlement === undefined) {
+          return {
+            ...event.returnValues,
+            identifier: this.hexToUtf8(event.returnValues.identifier),
+            ancillaryData: event.returnValues.ancillaryData ? event.returnValues.ancillaryData : "0x",
+          };
+        }
+      })
+      .filter(isDefined);
 
     // Filter proposals based on their expiration timestamp:
-    const isExpired = (proposal: OptimisticOracleWeb3Events.ProposePrice["returnValues"]) => {
-      return Number(proposal.expirationTimestamp) <= Number(currentTime);
+    const isExpired = (proposal: ProposePriceReturnValues): boolean => {
+      if (this.oracleType === "SkinnyOptimisticOracle") {
+        return (
+          Number(
+            ((((proposal as unknown) as SkinnyProposePrice["returnValues"]).request as unknown) as SkinnyRequest)
```

Comment:
nothing to change here but I'm not a fan of these `unknown` type casts all over the place like this. wish there was a cleaner pattern we could follow.